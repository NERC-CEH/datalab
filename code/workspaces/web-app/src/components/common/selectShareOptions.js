import React from 'react';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core';

/* eslint-disable max-len */
const CONTENT = {
  notebook: {
    projectDescription: 'Anyone in this project can view and interact with this notebook.',
    privateDescription: 'Only you can view and interact with this notebook.',
    projectWarning: 'NOTE: This option will allow multiple users to view and edit notebooks at the same time. As real-time collaboration is not currently supported this may result in corruption, hence this option should be chosen primarily for notebooks that are being viewed/run rather than edited.',
  },
  site: {
    privateDescription: 'Only you can can view and interact with this site.',
    projectDescription: 'Anyone in this project can view and interact with this site.',
    publicDescription: 'Anyone can view and interact with this site.',
    publicWarning: 'NOTE: This will make the site available (via URL) to anyone including those external to the DataLabs platform. Hence please ensure any security considerations have been implemented before proceeding.',
  },
};
/* eslint-disable max-len */

const useStyles = makeStyles(theme => ({
  select: {
    whiteSpace: 'normal',
    maxWidth: 500,
    marginTop: theme.spacing(1),
    marginBottom: theme.spacing(1),
  },
  title: {
    fontWeight: 500,
  },
  description: {
    color: theme.typography.colorLight,
  },
  warning: {
    marginTop: theme.spacing(0.5),
    color: 'hsl(0, 40%, 40%)',
  },
}));

const SelectRichComponent = ({ type, title }) => {
  const classes = useStyles();
  const option = title.toLowerCase();

  return (
    <div className={classes.select}>
      <Typography variant="body1" className={classes.title}>{title}</Typography>
      <Typography variant="body1" className={classes.description}>{CONTENT[type][`${option}Description`]}</Typography>
      { (`${option}Warning` in CONTENT[type]) && <Typography variant="body1" className={classes.warning}>{CONTENT[type][`${option}Warning`]}</Typography> }
    </div>
  );
};

const notebookSharingOptions = [
  {
    text: <SelectRichComponent type="notebook" title="Private"/>,
    value: 'private',
  },
  {
    text: <SelectRichComponent type="notebook" title="Project"/>,
    value: 'project',
  },
];

const siteVisibilityOptions = [
  {
    text: <SelectRichComponent type="site" title="Private"/>,
    value: 'private',
  },
  {
    text: <SelectRichComponent type="site" title="Project"/>,
    value: 'project',
  },
  {
    text: <SelectRichComponent type="site" title="Public"/>,
    value: 'public',
  },
];

export { notebookSharingOptions, siteVisibilityOptions };
