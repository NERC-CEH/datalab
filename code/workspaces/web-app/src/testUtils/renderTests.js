import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider as MuiThemeProvider } from '@mui/material/styles';
import theme from '../theme';

const AllTheProviders = ({ children }) => (
    <MuiThemeProvider theme={theme}>
        {children}
    </MuiThemeProvider>
);

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
