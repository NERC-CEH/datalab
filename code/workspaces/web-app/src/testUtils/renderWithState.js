import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';
import promise from 'redux-promise-middleware';
import {
  PROMISE_TYPE_PENDING,
  PROMISE_TYPE_SUCCESS,
  PROMISE_TYPE_FAILURE,
} from '../actions/actionTypes';

const promiseTypeSuffixes = [PROMISE_TYPE_PENDING, PROMISE_TYPE_SUCCESS, PROMISE_TYPE_FAILURE];

const TEST_USER = 'test-user';

export const buildDefaultTestState = () => ({
  authentication: {
    identity: {
      sub: TEST_USER,
    },
    permissions: {
      fetching: { inProgress: false },
      value: [],
    },
  },
  projectUsers: {
    fetching: { inProgress: false },
    value: [
      { userId: TEST_USER, role: 'admin' },
    ],
  },
});

export const renderWithState = (
  initialState,
  ComponentToRender,
  props = {},
  renderOptions = {},
) => {
  const mockStore = configureStore([promise({ promiseTypeSuffixes })]);
  const store = mockStore(initialState);

  const wrapper = render(
    <Provider store={store}>
      <ComponentToRender {...props} />
    </Provider>,
    renderOptions,
  );

  return {
    wrapper,
    store,
  };
};
