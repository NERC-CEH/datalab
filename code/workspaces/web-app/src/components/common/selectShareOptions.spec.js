import { render } from '../../testUtils/renderTests';
import { notebookSharingOptions, siteVisibilityOptions } from './selectShareOptions';

describe('NotebookSharingOptions-private', () => {
  it('correctly renders correct snapshot', () => {
    expect(render(notebookSharingOptions[0].text).container).toMatchSnapshot();
  });
});

describe('NotebookSharingOptions-project', () => {
  it('correctly renders correct snapshot', () => {
    expect(render(notebookSharingOptions[1].text).container).toMatchSnapshot();
  });
});

describe('SiteVisibilityOptions-private', () => {
  it('correctly renders correct snapshot', () => {
    expect(render(siteVisibilityOptions[0].text).container).toMatchSnapshot();
  });
});

describe('SiteVisibilityOptions-project', () => {
  it('correctly renders correct snapshot', () => {
    expect(render(siteVisibilityOptions[1].text).container).toMatchSnapshot();
  });
});

describe('SiteVisibilityOptions-public', () => {
  it('correctly renders correct snapshot', () => {
    expect(render(siteVisibilityOptions[2].text).container).toMatchSnapshot();
  });
});
