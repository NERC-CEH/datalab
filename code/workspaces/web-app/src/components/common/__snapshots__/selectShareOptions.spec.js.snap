// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotebookSharingOptions-private correctly renders correct snapshot 1`] = `
<div>
  <div
    class="makeStyles-select-1"
  >
    <p
      class="MuiTypography-root makeStyles-title-2 MuiTypography-body1"
    >
      Private
    </p>
    <p
      class="MuiTypography-root makeStyles-description-3 MuiTypography-body1"
    >
      Only you can view and interact with this notebook.
    </p>
  </div>
</div>
`;

exports[`NotebookSharingOptions-project correctly renders correct snapshot 1`] = `
<div>
  <div
    class="makeStyles-select-35"
  >
    <p
      class="MuiTypography-root makeStyles-title-36 MuiTypography-body1"
    >
      Project
    </p>
    <p
      class="MuiTypography-root makeStyles-description-37 MuiTypography-body1"
    >
      Anyone in this project can view and interact with this notebook.
    </p>
    <p
      class="MuiTypography-root makeStyles-warning-38 MuiTypography-body1"
    >
      NOTE: This option will allow multiple users to view and edit notebooks at the same time. As real-time collaboration is not currently supported this may result in corruption, hence this option should be chosen primarily for notebooks that are being viewed/run rather than edited.
    </p>
  </div>
</div>
`;

exports[`SiteVisibilityOptions-private correctly renders correct snapshot 1`] = `
<div>
  <div
    class="makeStyles-select-69"
  >
    <p
      class="MuiTypography-root makeStyles-title-70 MuiTypography-body1"
    >
      Private
    </p>
    <p
      class="MuiTypography-root makeStyles-description-71 MuiTypography-body1"
    >
      Only you can can view and interact with this site.
    </p>
  </div>
</div>
`;

exports[`SiteVisibilityOptions-project correctly renders correct snapshot 1`] = `
<div>
  <div
    class="makeStyles-select-103"
  >
    <p
      class="MuiTypography-root makeStyles-title-104 MuiTypography-body1"
    >
      Project
    </p>
    <p
      class="MuiTypography-root makeStyles-description-105 MuiTypography-body1"
    >
      Anyone in this project can view and interact with this site.
    </p>
  </div>
</div>
`;

exports[`SiteVisibilityOptions-public correctly renders correct snapshot 1`] = `
<div>
  <div
    class="makeStyles-select-137"
  >
    <p
      class="MuiTypography-root makeStyles-title-138 MuiTypography-body1"
    >
      Public
    </p>
    <p
      class="MuiTypography-root makeStyles-description-139 MuiTypography-body1"
    >
      Anyone can view and interact with this site.
    </p>
    <p
      class="MuiTypography-root makeStyles-warning-140 MuiTypography-body1"
    >
      NOTE: This will make the site available (via URL) to anyone including those external to the DataLabs platform. Hence please ensure any security considerations have been implemented before proceeding.
    </p>
  </div>
</div>
`;
